src/store/contact-book/contact-book-reducers.ts

```diff
@@ -14,13 +14,9 @@ export const contactBookReducers = createReducer<ContactBookState>(contactBookIn
     state.contacts = [...state.contacts, payload];
   });
   builder.addCase(editContactAction, (state, { payload }) => {
-    const contactIndex = state.contacts.findIndex(contact => contact.publicKeyHash === payload.publicKeyHash);
-
-    if (contactIndex !== -1) {
-      const contactsCopy = [...state.contacts];
-      contactsCopy[contactIndex] = payload;
-      state.contacts = contactsCopy;
-    }
+    const contactsCopy = [...state.contacts];
+    contactsCopy[payload.index] = payload.contact;
```

Comment:
why do you pass `index` everywhere?  `contact.publicKeyHash` is not unique?

Reply:
In case of editing it is not unique